src/compiler/crystal/semantic/bindings.cr

```diff
@@ -8,44 +8,12 @@ module Crystal
   # bytes) vs 2 pointers (16 bytes). The other downside is that since this is a
   # struct, we need to be careful with mutation.
   struct SmallNodeList
-    include Indexable(ASTNode)
+    include Enumerable(ASTNode)
 
     @first : ASTNode?
     @second : ASTNode?
     @tail : Array(ASTNode)?
 
-    def size
-      size = 0
-      size += 1 unless @first.nil?
-      size += 1 unless @second.nil?
-      if tail = @tail
-        size += tail.size
-      end
-      size
-    end
```

Comment:
**suggestion:** We should keep the implementation of `#size`. The parent implementation of `Enumerable#size` iterates all elements which would be less efficient than adding the known tail size.